packages/react-data-grid/src/masks/InteractionMasks.tsx

```diff
@@ -0,0 +1,750 @@
+  static displayName = 'InteractionMasks';
+
+  readonly state: Readonly<State> = {
+    selectedPosition: {
+      idx: -1,
+      rowIdx: -1
+    },
+    selectedRange: {
+      topLeft: {
+        idx: -1, rowIdx: -1
+      },
+      bottomRight: {
+        idx: -1, rowIdx: -1
+      },
+      startCell: null,
+      cursorCell: null,
+      isDragging: false
+    },
+    copiedPosition: null,
+    draggedPosition: null,
+    editorPosition: null,
+    isEditorEnabled: false,
+    firstEditorKeyPress: null
+  };
+
+  private readonly selectionMask = React.createRef<HTMLDivElement>();
+  private readonly copyMask = React.createRef<HTMLDivElement>();
+
+  private unsubscribeSelectCell: null | (() => void) = null;
+  private unsubscribeSelectStart: null | (() => void) = null;
+  private unsubscribeSelectUpdate: null | (() => void) = null;
+  private unsubscribeSelectEnd: null | (() => void) = null;
+  private unsubscribeDragEnter: null | (() => void) = null;
+
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    const { selectedPosition, isEditorEnabled } = this.state;
+    const { selectedPosition: prevSelectedPosition, isEditorEnabled: prevIsEditorEnabled } = prevState;
+    const isSelectedPositionChanged = selectedPosition !== prevSelectedPosition && (selectedPosition.rowIdx !== prevSelectedPosition.rowIdx || selectedPosition.idx !== prevSelectedPosition.idx);
+    const isEditorClosed = isEditorEnabled !== prevIsEditorEnabled && !isEditorEnabled;
+
+    if (isSelectedPositionChanged) {
+      // Call event handlers if selected cell has changed
+      const { onCellSelected, onCellDeSelected } = this.props;
+      if (onCellDeSelected && this.isCellWithinBounds(prevSelectedPosition)) {
+        onCellDeSelected({ ...prevSelectedPosition });
+      }
+
+      if (onCellSelected && this.isCellWithinBounds(selectedPosition)) {
+        onCellSelected({ ...selectedPosition });
+      }
+    }
+
+    if ((isSelectedPositionChanged && this.isCellWithinBounds(selectedPosition)) || isEditorClosed) {
+      this.focus();
+    }
+  }
+
+  componentDidMount() {
+    const { eventBus, enableCellAutoFocus } = this.props;
+    this.unsubscribeSelectCell = eventBus.subscribe(EventTypes.SELECT_CELL, this.selectCell);
+    this.unsubscribeSelectStart = eventBus.subscribe(EventTypes.SELECT_START, this.onSelectCellRangeStarted);
+    this.unsubscribeSelectUpdate = eventBus.subscribe(EventTypes.SELECT_UPDATE, this.onSelectCellRangeUpdated);
+    this.unsubscribeSelectEnd = eventBus.subscribe(EventTypes.SELECT_END, this.onSelectCellRangeEnded);
+    this.unsubscribeDragEnter = eventBus.subscribe(EventTypes.DRAG_ENTER, this.handleDragEnter);
+
+    if (enableCellAutoFocus && this.isFocusedOnBody()) {
+      this.selectFirstCell();
+    }
+  }
+
+  componentWillUnmount() {
+    this.unsubscribeSelectCell!();
+    this.unsubscribeSelectStart!();
+    this.unsubscribeSelectUpdate!();
+    this.unsubscribeSelectEnd!();
+    this.unsubscribeDragEnter!();
```

Comment:
We can do something like the following.

```
unsubscribeHandlers: (() => void)[] = [];

componentDidMount() {
   this.unsubscribeHandlers = [
     eventBus.subscribe(EventTypes.SELECT_START, this.onSelectCellRangeStarted),
     ...
   ];
}

componentWillUnmount() {
   this.unsubscribeHandlers.forEach(h => h());
}
```

wdyt?

Reply:
Done.